test/proxyprotocol/proxyprotocol_test.go

```diff
@@ -91,6 +91,77 @@ func TestProxyProtocol(t *testing.T) {
 	assert.Equal(t, resp.StatusCode, http.StatusOK)
 }
 
+// TestProxyProtocolWithSNI verifies that the kourier is configured with proxy protocol and SNI.
+func TestProxyProtocolWithSNI(t *testing.T) {
+	ctx, clients := context.Background(), test.Setup(t)
+	privateHostname := test.ObjectNameForTest(t) + "." + test.ServingNamespace + ".svc." + test.NetworkingFlags.ClusterSuffix
+	name, port, _ := ingress.CreateProxyService(ctx, t, clients, privateHostname, "kourier-internal.kourier-system.svc.cluster.local")
```

Comment:
`kourier-internal` should be replaced with `InternalServiceName` defined in `pkg/config/config.go`.

`kourier-system` also should be replaced with constant value. `test/ha/kourier_gateway_test.go` already has the value so we should move the const value somewhere in common place?